openSearchLane.go

```diff
@@ -246,242 +220,7 @@ func (osl *openSearchLane) Stats() (stats OslStats) {
 }
 
 func (osl *openSearchLane) SetEmergencyHandler(emergencyFn OslEmergencyFn) {
-	osl.openSearchConnection.laneMu.Lock()
-	defer osl.openSearchConnection.laneMu.Unlock()
+	osl.mu.Lock()
+	defer osl.mu.Unlock()
 	osl.openSearchConnection.emergencyFn = emergencyFn
```

Comment:
It's inappropriate for openSearchLane to modify the members of openSearchConnection directly. In this case the lane mutex is claimed, but that can collide with a derived lane doing the same thing.

All writes to openSearchConnection need to be done in openSearchConnection.go, under openSearchConnection's mutex.